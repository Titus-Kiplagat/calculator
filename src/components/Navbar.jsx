import { Link } from 'react-router-dom';

const Navbar = () => (
  <nav className="sticky top-0 select-none bg-gray-500 lg:flex lg:items-stretch w-full md:flex-row">
    <div className="flex justify-center flex-no-shrink items-stretch h-12">
      <Link to="/" className="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark text-2xl font-black uppercase">Math Calculator</Link>
    </div>
    <div className="lg:flex lg:items-stretch lg:flex-no-shrink lg:flex-grow">
      <div className="flex justify-around lg:items-stretch  ml-auto">
        <Link to="home" className="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark">Home</Link>
        <Link to="calculator" className="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark">Calculator</Link>
        <Link to="quote" className="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark">Quote</Link>
      </div>
    </div>
  </nav>
);

export default Navbar;
