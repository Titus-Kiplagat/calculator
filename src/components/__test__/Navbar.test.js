import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../Navbar';

describe('Navbar', () => {
  it('Navbar renders correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('Navbar contains links to Home, Calculator, and Quote', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    expect(getByText('Math Calculator')).toBeInTheDocument();
    expect(getByText('Home')).toBeInTheDocument();
    expect(getByText('Calculator')).toBeInTheDocument();
    expect(getByText('Quote')).toBeInTheDocument();
  });
});
