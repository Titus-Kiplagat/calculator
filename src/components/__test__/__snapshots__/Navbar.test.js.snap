// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="sticky top-0 select-none bg-gray-500 lg:flex lg:items-stretch w-full"
  >
    <div
      class="flex flex-no-shrink items-stretch h-12"
    >
      <a
        class="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark text-2xl font-black uppercase"
        href="/"
      >
        Math Calculator
      </a>
    </div>
    <div
      class="lg:flex lg:items-stretch lg:flex-no-shrink lg:flex-grow"
    >
      <div
        class="lg:flex lg:items-stretch lg:justify-end ml-auto"
      >
        <a
          class="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark"
          href="/home"
        >
          Home
        </a>
        <a
          class="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark"
          href="/calculator"
        >
          Calculator
        </a>
        <a
          class="flex-no-grow flex-no-shrink relative py-2 px-4 leading-normal text-white no-underline flex items-center hover:bg-grey-dark"
          href="/quote"
        >
          Quote
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
